rust/osm2lanes/src/transform/tags_to_lanes/mod.rs

```diff
@@ -244,11 +209,24 @@ impl RoadBuilder {
     #[allow(clippy::items_after_statements)]
     pub fn from(
         tags: &Tags,
-        oneway: Oneway,
-        lanes: &LanesScheme,
-        _locale: &Locale,
+        locale: &Locale,
         warnings: &mut RoadWarnings,
     ) -> Result<Self, RoadError> {
+        // Parse lane count schemas first.
+        let oneway = Oneway::from(
+            tags.is_any("oneway", &["yes", "-1"]) || tags.is("junction", "roundabout"),
+        );
+        let bus_lane_counts = BusLanesCount::from_tags(tags, locale, oneway, warnings)?;
+        let centre_turn_lanes = CentreTurnLaneScheme::new(tags, oneway, locale, warnings);
+        let lanes = Counts::new(
+            tags,
+            oneway,
+            &centre_turn_lanes,
+            &bus_lane_counts,
+            locale,
+            warnings,
+        );
```

Comment:
I love the idea that the road builder is responsible for all of this. It's pretty clearly the right spot.

Combined with "builder" in the name, it makes me think of something like this. The call graph becomes the things that describes the dependencies, which is interesting.

```rs
struct RoadBuilder {
    // ...
    pub oneway: Oneway,
    pub bus_lane_counts: BusLanesCount,
    pub centre_turn_lanes: CentreTurnLaneScheme,
    pub lane_counts: Counts,

    pub locale: Locale,
    warnings: RoadWarnings,
}

impl  RoadBuilder {
	pub fn from_tags(tags: &Tags, locale: &Locale) -> Result {
	      let road = Self::local_default(locale);
	      road.oneway = Oneway::from(tags.is_any("oneway", &["yes", "-1"]) || tags.is("junction", "roundabout"));
	      road.bus_lane_counts = BusLanesCount::from_tags(tags, &self)?;
	      road.centre_turn_lanes = CentreTurnLaneScheme::from_tags(tags, &self)?;
	      road.lane_counts = Counts::from_tags(tags, &self);
	}
	pub fn add_warning(...){...}
}

// ::default() for everything...
```